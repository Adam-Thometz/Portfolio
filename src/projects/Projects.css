.Projects {
  margin: 32px auto;
  animation: slide-up var(--duration) calc(var(--delay) * var(--i)) ease-in-out
    both;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  transition: opacity 0.3s ease-in-out;
  position: relative;
}

.Projects-project-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0 auto;
}

.project {
  transition: opacity 0.3s ease-in-out;
}

.display {
  transition: opacity 0.3s ease-in-out;
}

/* Styles for projects */

.art-buddy {
  background-color: rgba(77, 79, 94, 0.35);
  color: rgb(137, 235, 10);
  box-shadow: 0px 0px 20px 15px rgba(253, 77, 58, 0.5);
}

.art-buddy .Project-img img {
  border-color: rgb(255, 232, 26);
}

.reptifeed {
  background-color: rgba(69, 153, 0, 0.35);
  color: #d5ce3b;
  box-shadow: 0px 0px 20px 15px rgba(69, 153, 0, 0.5);
}

.eight-bit-clan {
  color: rgba(250, 250, 250);
  background-color: rgb(12, 17, 88, 0.35);
  box-shadow: 0px 0px 20px 15px rgba(250, 250, 250, 0.5);
}

.vesta-healing-psychotherapy {
  background-color: rgba(203, 155, 255, 0.35);
  color: rgba(255, 197, 162, 1);
  box-shadow: 0px 0px 20px 15px rgba(245, 129, 139, 0.5);
}

@media (max-width: 550px) {
  .Projects {
    flex-direction: column;
    align-items: center;
  }
}
