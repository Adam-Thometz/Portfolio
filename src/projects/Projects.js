import React from "react";
import projects from "./project-info";
import Project from "./Project";
import NavContact from "../navbar-contact/NavContact";
import './Projects.css'

const Projects = () => {
  return (
    <main className="Projects">
      <h1 className="Projects-header">Projects</h1>
      {projects.map(p => (
        <Project name={p.name} award={p.award} description={p.description} techStack={p.techStack} login={p.login} img={p.img} links={p.links} />
        ))}
      <NavContact />
    </main>
  );
};

export default Projects;