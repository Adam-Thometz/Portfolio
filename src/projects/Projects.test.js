import React from "react";
import { render } from "@testing-library/react";
import Projects from './Projects';

describe('Projects component', () => {
  it('renders without crashing', () => {
    render(<Projects />);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Projects />);
    expect(asFragment()).toMatchSnapshot();
  });
});