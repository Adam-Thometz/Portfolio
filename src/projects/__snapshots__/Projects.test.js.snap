// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="Projects display"
  >
    <figure
      class="ProjectCard art-buddy"
    >
      <figcaption>
        <h2>
          Art Buddy
        </h2>
        <p>
          Arts-based games for special education
        </p>
      </figcaption>
      <img
        alt="Art Buddy"
        src="art-buddy.webp"
      />
    </figure>
    <figure
      class="ProjectCard wedding-rsvp"
    >
      <figcaption>
        <h2>
          Wedding RSVP
        </h2>
        <p>
          A 20s themed wedding RSVP collector
        </p>
      </figcaption>
      <img
        alt="Wedding RSVP"
        src="wedding-rsvp.webp"
      />
    </figure>
    <figure
      class="ProjectCard reptifeed"
    >
      <figcaption>
        <h2>
          Reptifeed
        </h2>
        <p>
          A reptilian diet tracker
        </p>
      </figcaption>
      <img
        alt="Reptifeed"
        src="reptifeed.webp"
      />
    </figure>
    <figure
      class="ProjectCard eight-bit-clan"
    >
      <figcaption>
        <h2>
          8-Bit Clan
        </h2>
        <p>
          An 8-bit NFT maker
        </p>
      </figcaption>
      <img
        alt="8-Bit Clan"
        src="8-bit-clan.webp"
      />
    </figure>
    <figure
      class="ProjectCard musophile"
    >
      <figcaption>
        <h2>
          Musophile
        </h2>
        <p>
          An obscure music curator
        </p>
      </figcaption>
      <img
        alt="Musophile"
        src="musophile.webp"
      />
    </figure>
  </section>
</DocumentFragment>
`;
